data/governance/validations.go

```diff
@@ -0,0 +1,333 @@
+	}
+	if !reflect.DeepEqual(oldOptions.FirstLevelDomains, opt.FirstLevelDomains) {
+		return false, errors.New("first level domains cannot be changed")
+	}
+	ok, err := opt.PerBlockFees.CheckInRange(*minPerBlockFee, *maxPerBlockFee)
+	if err != nil || !ok {
+		return false, errors.Wrap(err, "Per Block Fee")
+	}
+	ok, err = opt.BaseDomainPrice.CheckInRange(*minBaseDomainPrice, *maxBaseDomainPrice)
+	if err != nil || !ok {
+		return false, errors.Wrap(err, "Base Domain Price")
+	}
+
+	return true, nil
+}
+
+func (st *Store) ValidateFee(opt *fees.FeeOption) (bool, error) {
+	oldOptions, err := st.GetFeeOption()
+	if err != nil {
+		return false, err
+	}
+	if !reflect.DeepEqual(opt.FeeCurrency, oldOptions.FeeCurrency) {
+		return false, errors.New("fee currency cannot be changed")
+	}
+	if !verifyRangeInt64(opt.MinFeeDecimal, minFeeDecimal, maxFeeDecimal) {
+		return false, errors.New("fee Decimal should be between 0 and 18")
+	}
+	return true, nil
+}
+
+func (st *Store) ValidateStaking(opt *delegation.Options) (bool, error) {
+	ok, err := opt.MinSelfDelegationAmount.CheckInRange(*minSelfDelegationAmount, *maxSelfDelegationAmount)
+	if err != nil || !ok {
+		return false, errors.Wrap(err, "Min Delegation Amount")
+	}
+	if !verifyRangeInt64(opt.TopValidatorCount, minValidatorCount, maxValidatorCount) {
+		return false, errors.New("validator count not within range")
+	}
+	if !verifyRangeInt64(opt.MaturityTime, minMaturityTime, maxMaturityTime) {
+		return false, errors.New("maturity time not with range")
+	}
+
+	return true, nil
+}
+
+func (st *Store) ValidateETH(opt *ethchain.ChainDriverOption) (bool, error) {
+	oldOptions, err := st.GetETHChainDriverOption()
+	if err != nil {
+		return false, err
+	}
+	if oldOptions.ContractABI != opt.ContractABI {
+		return false, errors.New("contract abi cannot be changed")
+	}
+	if oldOptions.ERCContractABI != opt.ERCContractABI {
+		return false, errors.New("erc contract abi cannot be changed")
+	}
+	if oldOptions.TotalSupply != opt.TotalSupply {
+		return false, errors.New("total supply cannot be changed")
+	}
+	if oldOptions.TotalSupplyAddr != opt.TotalSupplyAddr {
+		return false, errors.New("total supply address cannot be changed")
+	}
+	if !bytes.Equal(opt.ContractAddress.Bytes(), oldOptions.ContractAddress.Bytes()) {
+		return false, errors.New("smart contract address cannot be changed")
+	}
+	if !bytes.Equal(opt.ERCContractAddress.Bytes(), oldOptions.ERCContractAddress.Bytes()) {
+		return false, errors.New("ERC20 smart contract address cannot be changed")
+	}
+	if oldOptions.BlockConfirmation != opt.BlockConfirmation {
+		return false, errors.New("BlockConfirmations cannot be changed")
+	}
+	//if !verifyRangeInt64(opt.BlockConfirmation, minBlockConfirmation, maxBlockConfirmation) {
+	//	return false, errors.New("BlockConfirmations should be between 0 and 50")
+	//}
+	return true, nil
+}
+
+func (st *Store) ValidateBTC(opt *bitcoin.ChainDriverOption) (bool, error) {
```

Comment:
since we are not supporting BTC, I think we should just make this unchangeable.

Reply:
The update TX has  all the fields , I just verify whether the new fields are exactly equal to old or not. So it is unchangeable. If we want to remove this check altogether , we would need to fetch the BTC options from a separate place and not gov options ,is that what you are referring to ?

Reply:
it's okay as it right now, I got it